### ISSUE TITLE
Python 3.10 incompatible

### ISSUE THREAD

**Issue by frenck** [picklepete/pyicloud#368]
## The problem

Hi! 👋 

I've been working on upgrading Home Assistant to Python 3.10, and noticed this package isn't compatible with Python 3.10.

Python upgrade PR: <https://github.com/home-assistant/core/pull/59729>

Error from tests:

```
venv/lib/python3.10/site-packages/keyring/util/properties.py:1: in <module>
    from collections import Callable
E   ImportError: cannot import name 'Callable' from 'collections'

```

The `keyring` version used by this project is outdated and need to be bumped:

Currently it is using: `keyring>=8.0,<=9.3.1`

Could this be fixed/bump to be Python 3.10 compatible? 🙏  ❤️ 

Thanks!

../Frenck

## Environment

- pyiCloud release with the issue (`pip show pyicloud`): 
- Last working pyiCloud release (if known): 
- Service causing this issue: 
- Python version (`python -V`): 
- Operating environment (project deps/Docker/Windows/etc.): 

## Traceback/Error logs

```shell

```

## Additional information

PS: It would also be nice to look at: https://github.com/picklepete/pyicloud/issues/289
as this causes a dependency conflict @ Home Assistant.

PS2: Maybe bump & loosen the pinning of `tzlocal` dependency as well, as it also conflicts.

```
pyicloud 0.10.2 has requirement click<=7.1.1,>=6.0, but you have click 8.0.3.
pyicloud 0.10.2 has requirement tzlocal==2.0.0, but you have tzlocal 4.1.
```

**Comment by cdce8p** [comment]
Some additional information:
- The first fixed `keyring` version seems to be `15.0.0`.
https://github.com/jaraco/keyring/blob/v23.5.0/CHANGES.rst#1500
- A related Issue:
#290

**Comment by MartinHjelmare** [comment]
I think we can close here now. 1.0.0 supports Python 3.7 - Python 3.10.
